### PR TITLE
Expose BoundPattern.ConvertedType in IPatternOperation

### LINKED ISSUE #45681: Should expose BoundPattern.ConvertedType in IPatternOperation
Currently we have to call GetTypeInfo on syntax which returns the output type (ConvertedType) for a pattern node.

(maybe we should just call it  `OutputType`)

### PR DESCRIPTION
**Pull request by alrz**
Closes #45681

### CODE REVIEW

File: src/Compilers/Core/Portable/Operations/OperationInterfaces.xml [hunk 1]
```diff
@@ -2361,6 +2361,11 @@
         <summary>The input type to the pattern-matching operation.</summary>
       </Comments>
     </Property>
+    <Property Name="OutputType" Type="ITypeSymbol">
```
Comment (333fred) [inline src/Compilers/Core/Portable/Operations/OperationInterfaces.xml:2364]: I'll be honest: I don't know what this means. What concept is this actually trying to capture?

Comment (alrz) [reply src/Compilers/Core/Portable/Operations/OperationInterfaces.xml:2364]: this is spec'ed as the "narrowed type" or "output type" in pattern matching draft specification. it's most significant in `and` patterns where the output type of the LHS is the input type of the RHS. eg `obj is int and > 6`.

Comment (333fred) [reply src/Compilers/Core/Portable/Operations/OperationInterfaces.xml:2364]: That definition uses the term I'm confused on, so it doesn't really help :). Internally, we call this converted type. Is the type that the pattern input will be converted to? If so, `ConvertedType` is a significantly better name. The docs should also be more specific.

**Comment by gafter** [comment]
It might be a good idea to name it `NarrowedType` instead of `OutputType` to match the specification.

**Comment by alrz** [comment]
Test failures don't seem to be related.

File: src/Compilers/Core/Portable/Operations/OperationInterfaces.xml (hunk 1 above)
Comment (alrz) [reply src/Compilers/Core/Portable/Operations/OperationInterfaces.xml]: Sorry the name "output type" is not mentioned at all, I must misread something the first time.

Renamed.

**Comment by 333fred** [comment]
Thanks @alrz!

**Pull request merged**